### ISSUE TITLE
Q: how to disable console ouput expTable

### ISSUE THREAD

**Issue by dongyado** [emscripten-core/emscripten#11693]
hello, when load wasm module, console print a lot message like:
`
//
// This is an automatically generated file.
 // Do not edit.
//

const unsigned short expTable[] =
{
0x3c00, 0x3c00, 0x3c00, 0x3c00, 0x3c00, 0x3c00, 0x3c00, 0x3c00, 
...
`
it take a long time to finish print them, I think  informations is not help for debug or development. So how to disable it.

print call stack:

```
put_char (sxvideoengine.js:formatted:1831)
write (sxvideoengine.js:formatted:1779)
write (sxvideoengine.js:formatted:3204)
doWritev (sxvideoengine.js:formatted:4100)
_fd_write (sxvideoengine.js:formatted:5595)
_emscripten_receive_on_main_thread_js (sxvideoengine.js:formatted:10824)
func5740 (sxvideoengine.wasm:1)
kh (sxvideoengine.wasm:1)
nh (sxvideoengine.wasm:1)
Module._emscripten_main_thread_process_queued_calls (sxvideoengine.js:formatted:13152)
worker.onmessage (sxvideoengine.js:formatted:1358)
postMessage (async)
__emscripten_notify_thread_queue (sxvideoengine.js:1)
Ug (040569da:1)
ph (040569da:1)
Module._emscripten_run_in_main_runtime_thread_js (sxvideoengine.js:1)
_emscripten_proxy_to_main_thread_js (sxvideoengine.js:1)
_fd_write (sxvideoengine.js:1)
___wasi_fd_write (sxvideoengine.js:1)
(anonymous) (040569da:1)
(anonymous) (040569da:1)
(anonymous) (040569da:1)
(anonymous) (040569da:1)
(anonymous) (040569da:1)
(anonymous) (040569da:1)
Hh (040569da:1)
Module._main (sxvideoengine.js:1)
___call_main (sxvideoengine.js:1)
(anonymous) (040569da:1)
fi (040569da:1)
Module.dynCall_ii (sxvideoengine.js:1)
onmessage (sxvideoengine.worker.js:1)
```

my compilation options:
`
    set_target_properties(sxvideoengine PROPERTIES LINK_FLAGS "-s WASM=1 --bind -g0 -O3 -s USE_PTHREADS=1 -s PTHREAD_POOL_SIZE=8 -s USE_WEBGL2=1 -s FULL_ES3=1  -s PROXY_TO_PTHREAD=1 -s OFFSCREEN_FRAMEBUFFER=1 -s ALLOW_MEMORY_GROWTH=1 -s NO_EXIT_RUNTIME=1 -s TOTAL_MEMORY=64MB")
`

env:
emsdk: 1.39
compile:  fast-comp with cmake

**Comment by sbc100** [comment]
I don't think emscripten itself is printing that information.    Its seems that your program is writing the contents of a file to stdout for some reason.   Any idea where the file comes from?
 
If you build with `--profiling-funcs` you should see actual function names in the backtrace rather than `Ug` and `hp` (which looks like minified names).

**Comment by dongyado** [comment]
hello, I've added the --profiling-funcs to compile options. and the call stacktrace below

```
put_char (sxvideoengine.js:2521)
write (sxvideoengine.js:2472)
write (sxvideoengine.js:3996)
doWritev (sxvideoengine.js:4841)
_fd_write (sxvideoengine.js:6344)
_emscripten_receive_on_main_thread_js (sxvideoengine.js:11898)
__do_call (sxvideoengine.wasm:1)
_emscripten_current_thread_process_queued_calls (sxvideoengine.wasm:1)
_emscripten_main_thread_process_queued_calls (sxvideoengine.wasm:1)
Module._emscripten_main_thread_process_queued_calls (sxvideoengine.js:15074)
worker.onmessage (sxvideoengine.js:2002)
postMessage (async)
__emscripten_notify_thread_queue (sxvideoengine.js:7819)
_do_emscripten_dispatch_to_thread (099fe076:1000002)
_emscripten_run_in_main_runtime_thread_js (099fe076:1000002)
Module._emscripten_run_in_main_runtime_thread_js (sxvideoengine.js:15092)
_emscripten_proxy_to_main_thread_js (sxvideoengine.js:11861)
_fd_write (sxvideoengine.js:6341)
___wasi_fd_write (sxvideoengine.js:6354)
___stdio_write (099fe076:1000002)
___fwritex (099fe076:1000002)
_fwrite (099fe076:1000002)
__ZNSt3__211__stdoutbufIcE6xsputnEPKcl (099fe076:1000002)
__ZNSt3__216__pad_and_outputIcNS_11char_traitsIcEEEENS_19ostreambuf_iteratorIT_T0_EES6_PKS4_S8_S8_RNS_8ios_baseES4_ (099fe076:162006)
__ZNSt3__224__put_character_sequenceIcNS_11char_traitsIcEEEERNS_13basic_ostreamIT_T0_EES7_PKS4_m (099fe076:161280)
_main (099fe076:1000002)
Module._main (sxvideoengine.js:15248)
___call_main (sxvideoengine.js:2180)
___emscripten_thread_main (099fe076:1000002)
dynCall_ii (099fe076:1000002)
Module.dynCall_ii (sxvideoengine.js:15482)
onmessage (sxvideoengine.worker.js:129)
```
async post call stracktrace

```
put_char (sxvideoengine.js:2521)
write (sxvideoengine.js:2472)
write (sxvideoengine.js:3996)
doWritev (sxvideoengine.js:4841)
_fd_write (sxvideoengine.js:6344)
_emscripten_receive_on_main_thread_js (sxvideoengine.js:11898)
__do_call (sxvideoengine.wasm:1)
_emscripten_current_thread_process_queued_calls (sxvideoengine.wasm:1)
_emscripten_main_thread_process_queued_calls (sxvideoengine.wasm:1)
Module._emscripten_main_thread_process_queued_calls (sxvideoengine.js:15074)
worker.onmessage (sxvideoengine.js:2002)
postMessage (async)
__emscripten_notify_thread_queue (sxvideoengine.js:7819)
_do_emscripten_dispatch_to_thread (099fe076:1000002)
_emscripten_run_in_main_runtime_thread_js (099fe076:1000002)
Module._emscripten_run_in_main_runtime_thread_js (sxvideoengine.js:15092)
_emscripten_proxy_to_main_thread_js (sxvideoengine.js:11861)
_fd_write (sxvideoengine.js:6341)
___wasi_fd_write (sxvideoengine.js:6354)
___stdio_write (099fe076:1000002)
___fwritex (099fe076:1000002)
_fwrite (099fe076:1000002)
__ZNSt3__211__stdoutbufIcE6xsputnEPKcl (099fe076:1000002)
__ZNSt3__216__pad_and_outputIcNS_11char_traitsIcEEEENS_19ostreambuf_iteratorIT_T0_EES6_PKS4_S8_S8_RNS_8ios_baseES4_ (099fe076:162006)
__ZNSt3__224__put_character_sequenceIcNS_11char_traitsIcEEEERNS_13basic_ostreamIT_T0_EES7_PKS4_m (099fe076:161280)
_main (099fe076:1000002)
Module._main (sxvideoengine.js:15248)
___call_main (sxvideoengine.js:2180)
___emscripten_thread_main (099fe076:1000002)
dynCall_ii (099fe076:1000002)
Module.dynCall_ii (sxvideoengine.js:15482)
onmessage (sxvideoengine.worker.js:129)

```

and i can not find the output file in my own code, but I found something maybe can solve this:

 function write: function(stream, buffer, offset, length, position, canOwn)  in the stacktrace:
```
 write: function(stream, buffer, offset, length, position, canOwn) {
  if (length < 0 || position < 0) {
   throw new FS.ErrnoError(28);
  }
  if (FS.isClosed(stream)) {
   throw new FS.ErrnoError(8);
  }
  if ((stream.flags & 2097155) === 0) {
   throw new FS.ErrnoError(8);
  }
  if (FS.isDir(stream.node.mode)) {
   throw new FS.ErrnoError(31);
  }
  if (!stream.stream_ops.write) {
   throw new FS.ErrnoError(28);
  }
  if (stream.seekable && stream.flags & 1024) {
   FS.llseek(stream, 0, 2);
  }
  var seeking = typeof position !== "undefined";
  if (!seeking) {
   position = stream.position;
  } else if (!stream.seekable) {
   throw new FS.ErrnoError(70);
  }
  var bytesWritten = stream.stream_ops.write(stream, buffer, offset, length, position, canOwn);
  if (!seeking) stream.position += bytesWritten;
  try {
   if (stream.path && FS.trackingDelegate["onWriteToFile"]) FS.trackingDelegate["onWriteToFile"](stream.path);
  } catch (e) {
   err("FS.trackingDelegate['onWriteToFile']('" + stream.path + "') threw an exception: " + e.message);
  }
  return bytesWritten;
 }
```

 
** var bytesWritten = stream.stream_ops.write(stream, buffer, offset, length, position, canOwn);**

**at this line, I can get the buffer size is 67108864 = 64MB when I debug the code,
 so I think emscripten are print the initial memory data.** 

my compiled options contained ALLOW_MEMORY_GROWTH and TOTAL_MEMORY=64MB

all options:
```
set_target_properties(sxvideoengine PROPERTIES LINK_FLAGS "-s WASM=1 --bind -g0 -O3 -s USE_PTHREADS=1 -s PTHREAD_POOL_SIZE=8 -s USE_WEBGL2=1 -s FULL_ES3=1 -s PROXY_TO_PTHREAD=1 -s OFFSCREEN_FRAMEBUFFER=1 -s ALLOW_MEMORY_GROWTH=1 -s NO_EXIT_RUNTIME=1 -s TOTAL_MEMORY=64MB")
```

so I think emscripten print the initial memory data. and maybe there is way to disable the output.

**Comment by sbc100** [comment]
It looks to my like there is code in you main function that is writing to stdout, no?   emscripten on its own won't write to stdout like that.   

My googling for the string "const unsigned short expTable" it looks like this might be related to the freeimage library:

 https://stackoverflow.com/questions/614492/how-do-i-find-my-programs-main-function

Perhaps the freeimage library is being compiled and by mistake its utility programs are bring included the main function being called here is actually a utility program for freeimage that outputs this `expTable` data?    

Are you sure your main function is even running?

**Comment by dongyado** [comment]
whoa, thanks for remind me of  the 3-party library. It's werid that I have not googled with that string. sorry for that.
I'll write solutions for this issue.
Yes, I used freeimage and it depends on openEXR library. 

there are several files I found in openEXR which have main function:

**OpenEXR/Half/eLut.cpp
OpenEXR/Half/toFloat.cpp
OpenEXR/ilmimf/b44ExpLogTable.cpp**

cause freeimage using generate shell to generate Makefile, and the shell just included all files in openEXR.
there are options:
1. disable openEXR
**2. just remove the 3 files above or comment the main functions.**

thanks very much, hope this solution will help someone.
